src/lib/statistics/join_graph_statistics_cache.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <map>
+#include <unordered_map>
+
+#include "boost/dynamic_bitset.hpp"
+
+#include "expression/abstract_expression.hpp"
+
+namespace opossum {
+
+class TableStatistics;
+class JoinGraph;
+
+/**
+ * Cache of TableStatistics for LQPs consisting exclusively of JoinNodes, PredicateNodes and
+ * <Vertex Nodes>. That is, this is a cache for statistics associated with LQPs representing a subset of a JoinGraph.
+ *
+ * The key of the cache is a bitmask (with each bit representing a predicate or a vertex node).
+ *
+ * This cache exists primarily to aid the performance of the JoinOrderingRule.
+ * The JOR frequently request statistics for different plans consisting of the same set of Join and Scan predicates.
+ */
+class JoinGraphStatisticsCache {
+ public:
+  using Bitmask = boost::dynamic_bitset<>;
+  using VertexIndexMap = std::unordered_map<std::shared_ptr<AbstractLQPNode>, size_t>;
+  using PredicateIndexMap = ExpressionUnorderedMap<size_t>;
+
+  static JoinGraphStatisticsCache from_join_graph(const JoinGraph& join_graph);
```

Comment:
Added 
```
  // Creates a JoinGraphStatisticsCache with VertexIndexMap and PredicateIndexMap pointing to the vertices / predicates
  // in the JoinGraph
```

why is this not a "proper constructor"? Because, imo, a constructor should take the arguments that actually make up the data structure and not some other class from which they can be derived. I want to emphasize that there is actual work being done to get from JoinGraph to JoinGraphStatisticsCache.